app/main.py

```diff
@@ -1,8 +1,43 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+    instances = []
+
+    for person in people:
+        name = person["name"]
+        age = person["age"]
+
+        person_instance = Person(name, age)
+        instances.append(person_instance)
+
+    for person in people:
+        name = person["name"]
+
+        if "wife" in person or "husband" in person:
+
+            spouse_key = "wife" if "wife" in person else "husband"
+            spouse_name = person[spouse_key]
+
+            if spouse_name is not None:
```

Comment:
The check `if spouse_name is not None:` is redundant because the subsequent check `if spouse_name in Person.people:` will handle cases where `spouse_name` is `None` by returning `False`. You can remove this line to simplify the code.